src/ads_impl.cc

```diff
@@ -649,19 +649,47 @@ void AdsImpl::OnGetAdsForCategory(
   ShowAd(ad, category);
 }
 
-void AdsImpl::OnGetAdForSampleCategory(
+void AdsImpl::OnGetAdSampleBundle(
     const Result result,
-    const std::string& category,
-    const AdInfo& ad) {
+    const std::string& sample_bundle_json) {
   if (result == Result::FAILED) {
+    LOG(LogLevel::WARNING) << "Could not load sample bundle";
```

Comment:
without this change all of this logic would have to go in brave-core. Until we change the API to match SaveBundleState, etc.. I think this makes more sense than managing this logic in every client